test/user.test.js

```diff
@@ -2019,6 +2070,88 @@ describe('User', function() {
         return User.create({email: 'test@example.com', password: 'pass'})
           .then(u => user = u);
       }
+
+      it('is called over REST method /User/:id/verify', function() {
+        return User.create({email: 'bar@bat.com', password: 'bar'})
+          .then(user => {
+            return request(app)
+              .post('/test-users/' + user.pk + '/verify')
+              .expect('Content-Type', /json/)
+              // we already tested before that User.verify(id) works correctly
+              // having the remote method returning 204 is enough to make sure
+              // User.verify() was called successfully
+              .expect(204);
+          });
+      });
+
+      it('fails over REST method /User/:id/verify with invalid user id', function() {
+        return request(app)
+          .post('/test-users/' + 'invalid-id' + '/verify')
+          .expect('Content-Type', /json/)
+          .expect(404);
+      });
+    });
+
+    describe('User.verify(id)', function() {
+      it('triggers user identity verification', function() {
+        const verifyOptions = User.getVerifyOptions();
+
+        return User.create({email: 'bar@bat.com', password: 'bar'})
+        .then(user => {
+          return User.verify(user.pk, verifyOptions);
+        })
+        .then(res => {
+          // No need to redo all the tests already done in the user.verify()
+          // prototype method: just doing a partial check to confirm that the
+          // prototype method was called successfully.
+          expect(res.email.envelope).to.eql({
+            from: 'noreply@example.com',
+            to: ['bar@bat.com'],
+          });
+        });
+      });
+
+      it('throws when providing an invalid user id', function() {
+        return User.create({email: 'bar@bat.com', password: 'bar'})
+          .then(user => {
+            return User.verify('invalidUserId');
+          })
+          .then(
+            function onSuccess() {
+              throw new Error('User.verify() should have failed');
+            },
+            function onError(err) {
+              err = Object.assign({}, err);
+              expect(err).to.eql({
+                code: 'USER_NOT_FOUND',
+                statusCode: 404,
+              });
+            }
+          );
+      });
+
+      it('forwards the "options" argument', function() {
+        const options = {testFlag: true};
+        const verifyOptions = User.getVerifyOptions();
+
+        // backup original user.verify() method
+        const originalMethod = User.prototype.verify;
+
+        // injecting mockup user.verify() method to ease option forwarding test
+        User.prototype.verify = function(verifyOptions, options, cb) {
+          expect(options).to.eql({testFlag: true});
```

Comment:
If `User.prototype.verify` is not called at all, then the test will incorrectly pass.

A better solution is to store `options` to `observedOptions` and then check the value in a `.then` block in the main test flow below.